extensions_admin/pulp_docker/extensions/admin/manifest.py

```diff
@@ -0,0 +1,118 @@
+from gettext import gettext as _
+
+from pulp.client.commands import options
+from pulp.client.commands.criteria import DisplayUnitAssociationsCommand
+from pulp.client.commands.unit import UnitCopyCommand, UnitRemoveCommand
+
+from pulp_docker.common.models import Manifest, Blob
+
+DESC_COPY = _('copies manifests from one repository into another')
+DESC_REMOVE = _('remove manifests from a repository')
+DESC_SEARCH = _('search for manifests in a repository')
+FORMAT_ERR = _('The docker manifest formatter can not process %s units.')
+
+UNIT_ID_TEMPLATE = '%(digest)s'
+
+
+def get_formatter_for_type(type_id):
+    """
+    Returns a formatter that can be used to format the unit key
+    of a docker manifest or blob for display purposes.
+
+    :param type_id: A unit type ID.
+    :type type_id: str
+    :return: A formatter.
+    :rtype: callable
+    :raises ValueError: when the type_id is not supported.
+    """
+    if type_id in [Manifest.TYPE_ID, Blob.TYPE_ID]:
+        return lambda u: UNIT_ID_TEMPLATE % u
+    else:
+        raise ValueError(FORMAT_ERR % type_id)
+
+
+class ManifestSearchCommand(DisplayUnitAssociationsCommand):
+
+    def __init__(self, context):
+        """
+        :param context: A client context.
+        :type  context: pulp.client.extensions.core.ClientContext
+        """
+        super(ManifestSearchCommand, self).__init__(
+            name='search',
+            description=DESC_SEARCH,
+            method=self.run)
+        self.context = context
+        self.prompt = context.prompt
+
+    def run(self, **kwargs):
+        """
+        Print a list of all the manifests matching the search parameters.
+
+        :param kwargs: the search parameters for finding docker manifests
+        :type kwargs: dict
+        """
+        repo_id = kwargs.pop(options.OPTION_REPO_ID.keyword)
+        kwargs['type_ids'] = [Manifest.TYPE_ID]
+        reply = self.context.server.repo_unit.search(repo_id, **kwargs)
+        manifests = reply.response_body
+        self.prompt.render_document_list(manifests)
+
+
+class ManifestCopyCommand(UnitCopyCommand):
+
+    def __init__(self, context):
+        """
+        :param context: A client context.
+        :type  context: pulp.client.extensions.core.ClientContext
+        """
+        super(ManifestCopyCommand, self).__init__(
+            context,
+            name='copy',
+            description=DESC_COPY,
```

Comment:
No need to change it, but I've always wondered why the CLI code has this pattern of making the description a CONSTANT that is only used in one place instead of just writing the description here. I mean, I guess it gives us one place to look in the code for all the descriptions, so I guess that's fine ☺